apps/prairielearn/src/pages/instructorCourseAdminSettings/instructorCourseAdminSettings.ts

```diff
@@ -18,37 +19,79 @@ router.get(
     const courseInfoExists = await fs.pathExists(
       path.join(res.locals.course.path, 'infoCourse.json'),
     );
+    const availableTimezones = await getAvailableTimezones();
 
     res.send(
-      InstructorCourseAdminSettings({ resLocals: res.locals, coursePathExists, courseInfoExists }),
+      InstructorCourseAdminSettings({
+        resLocals: res.locals,
+        coursePathExists,
+        courseInfoExists,
+        availableTimezones,
+      }),
     );
   }),
 );
 
-router.post('/', (req, res, next) => {
-  if (!res.locals.authz_data.has_course_permission_edit || res.locals.course.example_course) {
-    return next(
-      error.make(403, 'Access denied (must be course editor and must not be example course)'),
-    );
-  }
-
-  if (req.body.__action === 'add_configuration') {
-    const editor = new CourseInfoEditor({
-      locals: res.locals,
-    });
-    editor.canEdit((err) => {
-      if (ERR(err, next)) return;
-      editor.doEdit((err, job_sequence_id) => {
-        if (ERR(err, (e) => logger.error('Error in doEdit()', e))) {
-          res.redirect(res.locals.urlPrefix + '/edit_error/' + job_sequence_id);
-        } else {
-          res.redirect(req.originalUrl);
-        }
+router.post(
+  '/',
+  asyncHandler(async (req, res, next) => {
+    if (!res.locals.authz_data.has_course_permission_edit || res.locals.course.example_course) {
+      return next(
+        error.make(403, 'Access denied (must be course editor and must not be example course)'),
+      );
+    }
+
+    if (req.body.__action === 'update_configuration') {
+      const courseInfo = JSON.parse(
+        await fs.readFileSync(path.join(res.locals.course.path, 'infoCourse.json'), 'utf8'),
+      );
+      courseInfo.name = req.body.short_name;
+      courseInfo.title = req.body.title;
+      courseInfo.timezone = req.body.display_timezone;
```

Comment:
I think we need to be a little careful handling this column. Currently some courses are relying on the default that's set for this value during syncing, and might also rely on inherited values from the institution. I think we want to do the following:

- If `timezone` is specified in the file and is a string, write the POSTed timezone back to the file.
- If `timezone` is not present in the file or is not a string, only write the timezone into the file if it's different from `res.locals.course.display_timezone`.

Reply:
Actually, I take it back! I think it makes sense to always explicitly write the timezone back to the file.

Reply:
Always explicitly writing the timezone back to the file. Also, as of [79f4f05](https://github.com/PrairieLearn/PrairieLearn/pull/9581/commits/79f4f05b8d78db39a94b29e232a22bdc4c552b34) we are now writing the default institution timezone when creating the file.